### PR TITLE
Align XPU unit tests

### LINKED ISSUE #2779: Accuracy failures in logspace op
Cases:
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_comprehensive_logspace_tensor_overload_xpu_int16~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_comprehensive_logspace_tensor_overload_xpu_int32~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_comprehensive_logspace_tensor_overload_xpu_int64~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_comprehensive_logspace_xpu_int16~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_comprehensive_logspace_xpu_int32~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_comprehensive_logspace_xpu_int64~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_quick_logspace_tensor_overload_xpu_int16~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_quick_logspace_tensor_overload_xpu_int32~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_quick_logspace_tensor_overload_xpu_int64~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_quick_logspace_xpu_int16~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_quick_logspace_xpu_int32~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_decomp_xpu.TestDecompXPU,test_quick_logspace_xpu_int64~~

Case failing with one-off error when running with integer support for some inputs.

### PR DESCRIPTION
**Pull request by Silv3S**
This PR groups XPU-related unit test skips and tolerance adjustments.

Fixes:
https://github.com/intel/torch-xpu-ops/issues/1973
https://github.com/intel/torch-xpu-ops/issues/2182
https://github.com/intel/torch-xpu-ops/issues/2779
https://github.com/intel/torch-xpu-ops/issues/2921

### CODE REVIEW

**Review by AKloniecki — APPROVED** [review]

**Review by chuanqi129 — APPROVED** [review]

**Review by daisyden — APPROVED** [review]
This PR is good for me.

**Comment by Silv3S** [comment]
Hi @mruberry, @malfet, @Skylion007
Could you please take a look? This patch affects only XPU backend and it already passed CI

**Comment by pbielak** [comment]
@Silv3S The commit `Skip fake autocast tests for linalg.pinv and pinverse on XPU` has been merged within its own PR - please rebase

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -14180,6 +14188,10 @@ def sample_inputs_abs(op_info, device, dtype, requires_grad, op_kwargs=None, **k
                             dtypes=(torch.int16, torch.int32, torch.int64), device_type="cuda"),
                DecorateInfo(unittest.expectedFailure, 'TestDecomp', 'test_comprehensive',
                             dtypes=(torch.int16, torch.int32, torch.int64), device_type="cuda"),
```
Comment (BBBela) [inline torch/testing/_internal/common_methods_invocations.py]:
```suggestion
               DecorateInfo(unittest.expectedFailure, 'TestDecomp', 'test_comprehensive',
                            dtypes=(torch.int16, torch.int32, torch.int64), device_type=("cuda", "xpu")),
```
As far as I know, we can now use the tuple as `device_type` in `DecorateInfo`, so we don't need to create separate entries, just change `"cuda"` to `("cuda", "xpu")`

Comment (Silv3S) [reply torch/testing/_internal/common_methods_invocations.py]: Sure, let's group them, to reduce duplication. Thanks for suggestions
